### PR TITLE
[#11] Setup initial project from Git template

### LINKED ISSUE #11: [Infrastructure] Setup initial project from Git template
## Why
Generate the project from our [Git template](https://github.com/nimblehq/git-template/) so that it can be beneficial from this template.

Also, introduce a new GitHub action that uses Terraform's `fmt` and `tfsec` to automatically check if there are any security & misconfiguration issues.

## Acceptance Criteria
- Generate the project from the Git Template
- Beneficial from the GitHub Actions & Wiki addon
- The linting step is configured to fail the build if linting errors are detected.
- Linting is triggered on every push to the repository and on every pull request.

### PR DESCRIPTION
**Pull request by sanG-github**
- Close #11

## What happened 👀

To start the DevOps IC, we should generate the codebase from our Git template.
- [x] Generate the project from our template
- [x] Add the GitHub actions that use `terraform fmt` and `tfsec` linters to run on `push` in every PR.

## Insight 📝

> [!NOTE]  
> CI action from Terraform Cloud will be passed on the next PR.

We added `checkov` recently in this PR, because it requires our workflows not run under `write-all` permission (by default), so we need to add the `read-all` permissions to our jobs.
```
Check: CKV2_GHA_1: "Ensure top-level permissions are not set to write-all"
	FAILED for resource: on(Publish Wiki)
Error: 	File: /.github/workflows/publish_wiki.yml:0-1
Check: CKV2_GHA_1: "Ensure top-level permissions are not set to write-all"
	FAILED for resource: on(Lint)
Error: 	File: /.github/workflows/lint_code.yml:0-1
```

## Proof Of Work 📹

| The action ran successfully                                                                                                                                                         	|
|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------	|
| Screenshot 2023-09-26 at 06 16 49 	|

### CODE REVIEW

**Review by Nihisil — APPROVED** [review]
:+1:

File: .github/workflows/lint_code.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  linting:
+    name: Lint Terraform files
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.11.0
+        with:
+          access_token: ${{ github.token }}
```
Comment (longnd) [inline .github/workflows/lint_code.yml]: We may not need this action by using the built-in [concurrency](https://docs.github.com/en/actions/using-jobs/using-concurrency) feature of Github action. Let's check if it is possible

File: .github/workflows/lint_code.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install Terraform
+        uses: hashicorp/setup-terraform@v2.0.3
+        with:
+          terraform_version: ${{ env.TERRAFORM_VERSION }}
+
+      - name: Run Terraform format
+        run: terraform fmt -recursive -check
```
Comment (longnd) [inline .github/workflows/lint_code.yml:33]: let's add another step to validate the syntax using Terraform [validate](https://developer.hashicorp.com/terraform/cli/commands/validate) command. That require you to init the terraform project first

```suggestion
      - name: Run Terraform format
        run: terraform fmt -recursive -check
        
      - name: Initialize Terraform
        run: terraform init
      
      - name: Validate Terraform configurations
        run: terraform validate -no-color    
```

File: .github/workflows/lint_code.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Run tfsec linter
+        id: tfsec
+        uses: aquasecurity/tfsec-action@v1.0.3
```
Comment (longnd) [inline .github/workflows/lint_code.yml:43]: Let's add a step to run [checkov](https://github.com/bridgecrewio/checkov) as in the requirement of the IC.

**Review by longnd — COMMENTED** [review]
Actually the step to setup the linter, publish wiki should be in another task (and PR) but since you already include them in this PR, let's process it

File: .github/workflows/lint_code.yml (hunk 1 above)
Comment (sanG-github) [reply .github/workflows/lint_code.yml]: I changed it to use `concurrency` instead in ad618c7 ✅

File: .github/workflows/lint_code.yml (hunk 3 above)
Comment (sanG-github) [reply .github/workflows/lint_code.yml:43]: I've added `checkov` in 4a5e4c5 ✅ Thanks for your reminder 🙏🏻

**Review by longnd — APPROVED** [review]

**Pull request merged**